tests/Resources/Page/OCPDashboard/DeploymentConfigs/DeploymentConfigs.robot

```diff
@@ -8,3 +12,19 @@ Start Rollout
     [Documentation]     Rollout a deployment in Openshift fetching the latest version
     [Arguments]  ${dc_name}  ${namespace}
     Run    oc rollout latest dc/${dc_name} -n ${namespace}
+
+Wait Until Rollout Is Started
```

Comment:
If it works we can merge it now to have it fixed today, but I think it could be improved in a future PR:
- It could be renamed to `Wait Until Rollout Is Finished` and wait until the pods are different and also until they are in  Running state (you can use the keyword `Get Pod Status`)
- Instead of looping you can use `Wait Until Keyword Succeeds` like [in this example](https://github.com/red-hat-data-services/ods-ci/blob/9e507824fc6a0268b8c4ca2f79f7e4813fa33e19/tests/Resources/Page/ODH/Prometheus/Prometheus.robot#L191)

Reply:
1. it is not waiting until rollout is finished. It just waits until it actually starts. By analyzing the failures, it came out that the rollout didn't start immediatedly and the `Wait Until JH Deployment Is Ready` (which do the checks on the "running" status etc) passed without actually waiting for it.
2. I don't think I can use it because I need two run two keyword: lines 22-24. I could use Run Keywords in combination with `Wait Until Keyword Succeeds` but I feel it is not very readable, but if you prefer I can do that

Reply:
I think it could be useful for other tests to verify that the rollout has been _finished_  (not just started), in order to not be necessary to implement an extra keyword for each deployment to check if it actually finished

I was thinking in something like this:

```
Verify Deployment Pods Have Been Updated
    [Arguments]     ${namespace}  ${label_selector}   ${comparison_fields}=['metadata.name']   ${previous_pods}
     ${current_pods}=    Oc Get    kind=Pod  namespace=${namespace}   label_selector=${label_selector}   fields=${comparison_fields}
     Should Not Be Equal As Strings    ${previous_pods}    ${current_pods}

Wait Until Rollout Is Finished
    [Documentation]     Wait until the old pods are replace with the new ones after a rollout is triggered
    [Arguments]     ${namespace}  ${label_selector}   ${comparison_fields}=['metadata.name']  ${retry}=5   ${retry_interval}=5s
    
    ${previous_pods}=    Oc Get    kind=Pod  namespace=${namespace}   label_selector=${label_selector}   fields=${comparison_fields}
    
    Wait Until Keyword Succeeds    ${retry}    ${retry_interval}
    ...    Verify Deployment Pods Have Been Updated    ${namespace}  ${label_selector}   ${comparison_fields} ${previous_pods}
    
    Wait Until Pod Status Is  ${namespace}  ${label_selector}   ${comparison_fields}   status=Running


```

Reply:
Agree, but in this particular case we wouldn't use this because Jupyterhub needs its own keyword to check if the deployment is ready (which automatically says if the rollout is finished).
So, I would avoid to implement in this PR a keyword which is not needed for this test case. Could we implement in the future when needed?

Reply:
Sure!

Reply:
I took a note about this. Thanks